### PR TITLE
executor: move test setup code from TestT to TestMain

### LINKED ISSUE #28440: Move executor test setup code from `TestT` to `TestMain`
## Enhancement
As currently tests under `executor` packages are using both `pingcap/check` and `testify`, we should move the test setup code in `TestT` to `main_test.go` to set up all of the tests written by both frameworks.
https://github.com/pingcap/tidb/blob/76267b21aee467f33426c2bdc1ccd3a5bd966b77/executor/executor_test.go#L89-L111
From my point of view, code related to `pingcap/check` can still be here, while code related to `testleak` can be replaced by `goleak` enabled in `TestMain`. All the other code will be better to move to `TestMain`.

### PR DESCRIPTION
**Pull request by unconsolable**
Signed-off-by: unconsolable <chenzhipeng2012@gmail.com>

### What problem does this PR solve?

Issue Number: close #28440 

### What is changed and how it works?

What's Changed:
- Move test setup code unrelated to `pingcap/check` from TestT to TestMain, which can set up tests written by both framework

### Release note

```release-note
None
```

### CODE REVIEW

**Comment by unconsolable** [comment]
/cc @tisonkun 
PTAL

File: executor/main_test.go [hunk 1]
```diff
@@ -15,14 +15,37 @@
...
 )
 
 func TestMain(m *testing.M) {
 	testbridge.WorkaroundGoCheckFlags()
+
+	logLevel := os.Getenv("log_level")
+	err := logutil.InitLogger(logutil.NewLogConfig(logLevel, logutil.DefaultLogFormat, "", logutil.EmptyFileLogConfig, false))
+	if err != nil {
+		panic(err.Error())
+	}
```
Comment (tisonkun) [inline executor/main_test.go:36]:
```suggestion
```

Redundant setup. We don't really depend on this. Can be removed.

**Review by wjhuang2016 — APPROVED** [review]

**Review by tisonkun — APPROVED** [review]
LGTM. Thanks for your help!

**Comment by tisonkun** [comment]
/assign @wjhuang2016 

Could you please help on merging?

**Pull request merged**